src/helpers.ts

```diff
@@ -184,9 +304,23 @@ export function loadOrCreateMinter(
   let filteredMinterContract = IFilteredMinterV2.bind(minterAddress);
 
   // values assigned in contract constructors
-  minter.minterFilter = filteredMinterContract
-    .minterFilterAddress()
-    .toHexString();
+  // @dev safely retrieve value, gracefully handle if it reverts
+  const minterFilterResult = filteredMinterContract.try_minterFilterAddress();
+  if (!minterFilterResult.reverted) {
+    minter.minterFilter = minterFilterResult.value.toHexString();
+  } else {
+    // if minterFilterAddress() reverts, then the minter is not as expected and
+    // we log warning, and assign to dummy MinterFilter entity at zero address
+    log.warning(
+      "[WARN] Minter at {} does not have a valid minter filter address",
+      [minterAddress.toHexString()]
+    );
+    const dummyMinterFilter = loadOrCreateSharedMinterFilter(
+      Address.zero(),
+      timestamp
+    );
+    minter.minterFilter = dummyMinterFilter.id;
+  }
```

Comment:
Very much a nit, but I find reducing boolean logic where possible is a bit more readable if we need to handle both the if and the else.
```suggestion
  if (minterFilterResult.reverted) {
    // if minterFilterAddress() reverts, then the minter is not as expected and
    // we log warning, and assign to dummy MinterFilter entity at zero address
    log.warning(
      "[WARN] Minter at {} does not have a valid minter filter address",
      [minterAddress.toHexString()]
    );
    const dummyMinterFilter = loadOrCreateSharedMinterFilter(
      Address.zero(),
      timestamp
    );
    minter.minterFilter = dummyMinterFilter.id;
  } else {
    minter.minterFilter = minterFilterResult.value.toHexString();
  }
```

Reply:
yeah this makes sense to me, thanks!